### ISSUE TITLE
Username Prompt

### ISSUE THREAD

**Issue by JackyChiu** [Hacker-Chatroom/Hacker-Chatroom-Server#2]
We left the project yesterday with the client waiting on the server to ask the user for the username through a web socket.

The only reason it would make sense to do this is for the server to store the username in say a `current users` collection in a database.

If we want to do this it should be with a REST endpoint instead of a socket asking for the username. But currently we only want the client to know the username so when it sends information, so it doesn't need to be implemented right away. 

TL;DR: Instead of entering your username when the socket asks you too, we should have the client ask for username then make a request to register your username

**Comment by BenEmdon** [comment]
:+1: with this

**Comment by vikrambombhi** [comment]
Agreed, but maybe we can have the client select a room before selecting a username or at least have the option to view the rooms first. Just because users might want to use a different username depending on the rooms available

**Comment by JackyChiu** [comment]
Should we do different usernames for each chatroom?

**Comment by vikrambombhi** [comment]
hmm idk maybe. I dont know how that would play out server side tho

**Comment by BenEmdon** [comment]
IMO we should do one username when we connect (it simplifies things for now) but later if you guys feel it's important we could make room usernames a stretch goal?

**Comment by vikrambombhi** [comment]
yea that makes sense

**Comment by BenEmdon** [comment]
![close/open](http://i.giphy.com/lbWE0pAlDwxBS.gif)

**Comment by vikrambombhi** [comment]
i tried
